spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -62,7 +59,7 @@
 
 	private static final Log LOGGER = LogFactory.getLog(PubSubTemplate.class);
 
-	private PubSubMessageConverter messageConverter = new JacksonPubSubMessageConverter();
+	private Converter<Object, ByteString> publishPayloadConverter;
```

Comment:
This fix is colliding with the #756 .
I believe we can stay with the current situation.

The `JacksonPubSubMessageConverter` can be resolved on demand when Spring Integration dependency is on class path.

You can add a classpath check static variable and only after that create a `JacksonPubSubMessageConverter` instance.
Something like this:
```
public static final boolean SPRING_INTEGRATION_PRESENT =
			ClassUtils.isPresent("org.springframework.integration.config.EnableIntegration",
					PubSubTemplate.class.getClassLoader());
```

Reply:
If we insist that `PubSubTemplate` should be independent of Spring Integration, this seems to compromise that position. Besides, why is conversion to JSON so tightly coupled to Spring Integration? It seems like Spring should just have this functionality somewhere in core.

Reply:
Strong +1 to being independent from SI in our core Pub/Sub module, especially if that functionally is not related to SI at all.

Reply:
That's easy. Just follow commit history: https://github.com/spring-cloud/spring-cloud-gcp/pull/659 !

I told there that we are going to be tied with Spring Integration...